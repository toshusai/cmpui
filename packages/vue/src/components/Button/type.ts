export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
