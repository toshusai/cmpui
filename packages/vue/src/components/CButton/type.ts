export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
