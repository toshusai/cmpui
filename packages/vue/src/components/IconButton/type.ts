export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
