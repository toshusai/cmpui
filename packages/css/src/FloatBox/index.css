.cmpui_float-box__root {
  background-color: var(--cmpui-bg-color);
  border: 1px solid var(--cmpui-border-color);
  box-shadow: var(--cmpui-float-shadow);

  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  font-size: 12px;
  height: fit-content;
  padding: 4px;
  position: relative;
  width: fit-content;
  z-index: 1000;
}
