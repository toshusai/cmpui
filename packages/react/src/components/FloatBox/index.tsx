import { forwardRef } from "react";

import { classNames } from "../../utils/classNames";

export const FloatBox = forwardRef<HTMLDivElement, React.ComponentProps<"div">>(
  function FloatBox({ className, ...props }, ref) {
    return (
      <div
        className={classNames("cmpui_float-box__root", className)}
        ref={ref}
        {...props}
      />
    );
  },
);
