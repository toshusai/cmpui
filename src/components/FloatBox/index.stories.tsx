import type { Meta, StoryObj } from "@storybook/react";
import { FloatBox } from ".";
import { ShortcutListItem } from "../../__stories__/ShortcutListItem";

const meta: Meta<typeof FloatBox> = {
  title: "Components/Float/FloatBox",
  component: FloatBox,
};

export default meta;
type Story = StoryObj<typeof FloatBox>;

export const Basic: Story = {
  render: function Render() {
    return (
      <FloatBox
        style={{
          width: 128,
        }}
      >
        <ShortcutListItem rounded size="S" shortcut="⌘ Z">
          Undo
        </ShortcutListItem>
        <ShortcutListItem rounded size="S" shortcut="⇧ ⌘ Z">
          Redo
        </ShortcutListItem>
        <ShortcutListItem rounded size="S" shortcut="⌘ X">
          Cut
        </ShortcutListItem>
        <ShortcutListItem rounded size="S" shortcut="⌘ C">
          Copy
        </ShortcutListItem>
        <ShortcutListItem rounded size="S" shortcut="⌘ V">
          Paste
        </ShortcutListItem>
      </FloatBox>
    );
  },
};
