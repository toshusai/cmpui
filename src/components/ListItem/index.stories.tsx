import type { Meta, StoryObj } from "@storybook/react";
import { List, ListItem } from ".";
import { FloatBox } from "../FloatBox";

const meta: Meta<typeof ListItem> = {
  title: "View/ListItem",
  component: ListItem,
};

export default meta;
type Story = StoryObj<typeof ListItem>;

export const Basic: Story = {
  render: function RenderBasic() {
    return (
      <List as="ol">
        <ListItem>Item1</ListItem>
        <ListItem>Item2</ListItem>
        <ListItem>Item3</ListItem>
      </List>
    );
  },
};

export const Selected: Story = {
  render: function RenderBasic() {
    return (
      <List as="ol">
        <ListItem selected>Item1</ListItem>
        <ListItem>Item2</ListItem>
        <ListItem>Item3</ListItem>
      </List>
    );
  },
};

export const Small: Story = {
  render: function RenderBasic() {
    return (
      <List as="ol">
        <ListItem size="S">Item1</ListItem>
        <ListItem size="S">Item2</ListItem>
        <ListItem size="S">Item3</ListItem>
      </List>
    );
  },
};

function CommandItem(props: {
  children?: React.ReactNode;
  shortcut?: string;
  selected?: boolean;
}) {
  return (
    <ListItem
      style={{
        width: 128,
      }}
      rounded
      selected={props.selected}
      size="S"
    >
      {props.children}
      {props.shortcut && (
        <span
          style={{
            marginLeft: "auto",
            fontSize: 10,
            color: props.selected ? "white" : "var(--cmpui-text2-color)",
          }}
        >
          {props.shortcut}
        </span>
      )}
    </ListItem>
  );
}

export const Commands: Story = {
  render: function RenderBasic() {
    return (
      <FloatBox>
        <List as="ol">
          <CommandItem shortcut="⌘ Z">Undo</CommandItem>
          <CommandItem shortcut="⇧ ⌘ Z">Redo</CommandItem>
          <CommandItem shortcut="⌘ X">Cut</CommandItem>
          <CommandItem selected shortcut="⌘ C">
            Copy
          </CommandItem>
          <CommandItem shortcut="⌘ V">Paste</CommandItem>
        </List>
      </FloatBox>
    );
  },
};
