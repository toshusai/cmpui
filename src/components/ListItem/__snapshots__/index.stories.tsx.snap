// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/View/ListItem Basic smoke-test 1`] = `
<ol class="cmpui_list-list__root">
  <li class="cmpui_list-item__root"
      data-size="M"
  >
    Item1
  </li>
  <li class="cmpui_list-item__root"
      data-size="M"
  >
    Item2
  </li>
  <li class="cmpui_list-item__root"
      data-size="M"
  >
    Item3
  </li>
</ol>
`;

exports[`Components/View/ListItem Commands smoke-test 1`] = `
<div class="cmpui_float-box__root">
  <ol class="cmpui_list-list__root">
    <li class="cmpui_list-item__root"
        data-size="S"
        data-rounded="true"
        style="width: 128px;"
    >
      Undo
      <span style="margin-left: auto; font-size: 10px; color: var(--cmpui-text2-color);">
        ⌘ Z
      </span>
    </li>
    <li class="cmpui_list-item__root"
        data-size="S"
        data-rounded="true"
        style="width: 128px;"
    >
      Redo
      <span style="margin-left: auto; font-size: 10px; color: var(--cmpui-text2-color);">
        ⇧ ⌘ Z
      </span>
    </li>
    <li class="cmpui_list-item__root"
        data-size="S"
        data-rounded="true"
        style="width: 128px;"
    >
      Cut
      <span style="margin-left: auto; font-size: 10px; color: var(--cmpui-text2-color);">
        ⌘ X
      </span>
    </li>
    <li class="cmpui_list-item__root"
        aria-selected="true"
        data-size="S"
        data-rounded="true"
        style="width: 128px;"
    >
      Copy
      <span style="margin-left: auto; font-size: 10px; color: white;">
        ⌘ C
      </span>
    </li>
    <li class="cmpui_list-item__root"
        data-size="S"
        data-rounded="true"
        style="width: 128px;"
    >
      Paste
      <span style="margin-left: auto; font-size: 10px; color: var(--cmpui-text2-color);">
        ⌘ V
      </span>
    </li>
  </ol>
</div>
`;

exports[`Components/View/ListItem Selected smoke-test 1`] = `
<ol class="cmpui_list-list__root">
  <li class="cmpui_list-item__root"
      aria-selected="true"
      data-size="M"
  >
    Item1
  </li>
  <li class="cmpui_list-item__root"
      data-size="M"
  >
    Item2
  </li>
  <li class="cmpui_list-item__root"
      data-size="M"
  >
    Item3
  </li>
</ol>
`;

exports[`Components/View/ListItem Small smoke-test 1`] = `
<ol class="cmpui_list-list__root">
  <li class="cmpui_list-item__root"
      data-size="S"
  >
    Item1
  </li>
  <li class="cmpui_list-item__root"
      data-size="S"
  >
    Item2
  </li>
  <li class="cmpui_list-item__root"
      data-size="S"
  >
    Item3
  </li>
</ol>
`;
